src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -449,7 +449,7 @@ public boolean doesMatchExist(User user, User otherUser) {
       // get users from match
       Pair<User, User> userPair = match.getUserPair();
       // compare users with users from match
-      if (userPair.getObj1() == user && userPair.getObj2() == otherUser) {
+      if (userPair.getObj1() == user && userPair.getObj2() == otherUser || userPair.getObj1() == otherUser && userPair.getObj2() == user) {
```

Comment:
I hope I understood correctly that the match should work in both directions..

Reply:
Yes, I will implement a comparator in the pair class later